bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,366 @@
+
+  Polynomial polynomial;
+  for (size_t i = 0; i < FIELD_ELEMENTS_PER_BLOB; i++)
+    bytes_to_bls_field(&polynomial[i], &blob[i * BYTES_PER_FIELD]);
+
+  KZGCommitment commitment;
+  blob_to_kzg_commitment(&commitment, polynomial, kzg_settings);
+
+  uint8_t commitment_bytes[BYTES_PER_COMMITMENT];
+  bytes_from_g1(commitment_bytes, &commitment);
+  return napi_typed_array_from_bytes(commitment_bytes, BYTES_PER_COMMITMENT, env);
+}
+
+// computeAggregateKzgProof: (blobs: Blob[], setupHandle: SetupHandle) => KZGProof;
+Napi::Value ComputeAggregateKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 2;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
+
+  auto blobs_count = blobs_param.Length();
+  auto polynomial = (Polynomial*)calloc(blobs_count, sizeof(Polynomial));
+
+  for (uint32_t blob_index = 0; blob_index < blobs_count; blob_index++) {
+    Napi::Value blob = blobs_param[blob_index];
+    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
+
+    for (size_t field_index = 0; field_index < FIELD_ELEMENTS_PER_BLOB; field_index++) {
+      bytes_to_bls_field(
+        &polynomial[blob_index][field_index],
+        &blob_bytes[field_index * BYTES_PER_FIELD]
+      );
+    }
+  }
+
+  KZGProof proof;
+  C_KZG_RET ret = compute_aggregate_kzg_proof(
+    &proof,
+    polynomial,
+    blobs_count,
+    kzg_settings
+  );
+  free(polynomial);
+
+  if (ret != C_KZG_OK) {
+     Napi::Error::New(env, "Failed to compute proof")
+      .ThrowAsJavaScriptException();
+    return env.Undefined();
+  };
+
+  uint8_t proof_bytes[BYTES_PER_PROOF];
+  bytes_from_g1(proof_bytes, &proof);
+  return napi_typed_array_from_bytes(proof_bytes, BYTES_PER_PROOF, env);
+}
+
+// verifyAggregateKzgProof: (blobs: Blob[], expectedKzgCommitments: KZGCommitment[], kzgAggregatedProof: KZGProof, setupHandle: SetupHandle) => boolean;
+Napi::Value VerifyAggregateKzgProof(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 4;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto blobs_param = info[0].As<Napi::Array>();
+  auto comittments_param = info[1].As<Napi::Array>();
+  auto proof_param = info[2].As<Napi::TypedArray>();
+  auto kzg_settings = info[3].As<Napi::External<KZGSettings>>().Data();
+
+  auto proof_bytes = proof_param.As<Napi::Uint8Array>().Data();
+  auto blobs_count = blobs_param.Length();
+  auto polynomial = (Polynomial*)calloc(blobs_count, sizeof(Polynomial));
+  auto commitments = (KZGCommitment*)calloc(blobs_count, sizeof(KZGCommitment));
```

Comment:
might as well wrap these in a vector to avoid manually (and possibly forgetting to) freeing the allocation.

Reply:
I tried for an hour and I don't think it's possible to make a `std::vector<Blob>`. https://stackoverflow.com/questions/16970624/cant-make-a-vector-of-fixed-size-arrays

Do you know how to do it? I think I want to just manage the memory manually.

Reply:
The error is 
```
error: object expression of non-scalar type 'unsigned char[131072]' cannot be used in a pseudo-destructor expression
```

Reply:
Oh interesting that `Blob` is typedef'd directly as a fixed-array. Yeah that won't work.
@xrchz perhaps we could redefine these types as structs like how blst does it [here](https://github.com/supranational/blst/blob/6382d67c72119d563975892ed49ba32e92d3d0da/bindings/blst.h#L60). This makes it more obvious that the types are arrays and lets the C++ compiler know that they're trivially copyable.

Reply:
Sure I can do it that way instead!

Reply:
Actually I don't really understand how to use those structs. Do you?

Reply:
I tried a version of this on a new branch, `struct_blob`, ptal.